### PR TITLE
Make modules/structs as a separate go mod

### LINKED ISSUE #177: SDK depends on Gitea
go-sdk depends currently on code.gitea.io/gitea that implicates that all dependencies of gitea are needed to for go-sdk. We should consider fixing this since it will increase the size of binaries of tools using the sdk.

### PR DESCRIPTION
**Pull request by lunny**
will fix https://github.com/go-gitea/go-sdk/issues/177

### CODE REVIEW

**Comment by lunny** [comment]
This should be ci skip, gitea build will depend `code.gitea.io/gitea/modules/structs`

File: modules/structs/attachment.go [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // license that can be found in the LICENSE file.
 
 package structs // import "code.gitea.io/gitea/modules/structs"
+
```
Comment (jonasfranz) [inline modules/structs/attachment.go:6]: why this change?

Comment (lunny) [reply modules/structs/attachment.go:6]: between `package` and `import` it should have a blank line but not.

**Review by sapk — APPROVED** [review]

**Review by jonasfranz — APPROVED** [review]

**Comment by jonasfranz** [comment]
@lunny please merge by ignoring drone
